test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -0,0 +1,455 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include <oneapi/dpl/execution>
+
+#include "support/test_config.h"
+
+#include "support/utils.h"
+
+#if _ENABLE_STD_RANGES_TESTING
+
+#include <oneapi/dpl/ranges>
+#include <span>
+#include <iostream>
+#include <vector>
+#include <typeinfo>
+#include <string>
+
+namespace test_std_ranges
+{;
+
+#if _ONEDPL_HETERO_BACKEND
+template<int call_id = 0>
+auto dpcpp_policy()
+{
+    auto exec = TestUtils::default_dpcpp_policy;
+    using Policy = decltype(exec);
+    return TestUtils::make_new_policy<TestUtils::new_kernel_name<Policy, call_id>>(TestUtils::default_dpcpp_policy);
+}
+#endif //_ONEDPL_HETERO_BACKEND
+
+auto host_policies() { return std::true_type{};}
+
+enum TestDataMode
```

Comment:
Minor comment: let's change `enum` -> `enum class` to avoid global namespace pollution.

Reply:
it is nit global, it is `namespace test_std_ranges` :)

Reply:
Ah, that's right. Then, it is not important at all. Do it if you would like to.